Pcap++/src/PcapLiveDevice.cpp

```diff
@@ -308,6 +340,34 @@ namespace pcpp
 		}
 #endif
 
+		if (config.timestampProvider)
+		{
+			ret = pcap_set_tstamp_type(pcap, timestampProviderMap(config.timestampProvider));
```

Comment:
According to [libpcap docs](https://www.tcpdump.org/manpages/libpcap-1.10.5/pcap-tstamp.7.html) we should first call [`pcap_list_tstamp_types`](https://www.tcpdump.org/manpages/libpcap-1.10.5/pcap_list_tstamp_types.3pcap.html) to figure out which type are possible for this device. I'm not sure what happens if we set a type that isn't supported...

Reply:
@seladb the behavior is described here:
 https://www.tcpdump.org/manpages/libpcap-1.10.5/pcap_set_tstamp_type.3pcap.html
 
 if I read it correctly, this means that:
   1. the error will be printed 
   2. the default timestamp type will be used. 
   
   Let me know if you prefer to add the check and e.g. throw an exception if the timestamp is not supported

Reply:
Thanks @vcomito-apexai ! I think it's ok to log an error

Reply:
added @seladb